torch/lib/THD/master_worker/worker/dispatch/TensorMath.cpp

```diff
@@ -0,0 +1,173 @@
+
+static void tensorDiag(rpc::RPCMessage& raw_message) {
+  Tensor *r = unpackRetrieveTensor(raw_message);
+  Tensor *t = unpackRetrieveTensor(raw_message);
+  int k = unpackInteger(raw_message);
+  finalize(raw_message);
+  r->diag(*t, k);
+}
+
+static void tensorEye(rpc::RPCMessage& raw_message) {
+  Tensor *tensor = unpackRetrieveTensor(raw_message);
+  long n = unpackInteger(raw_message);
+  long m = unpackInteger(raw_message);
+  finalize(raw_message);
+  tensor->eye(n, m);
+}
+
+static void tensorRange(rpc::RPCMessage& raw_message) {
+  Tensor *r = unpackRetrieveTensor(raw_message);
+  Type type = peekType(raw_message);
+  if (isInteger(type)) {
+    auto xmin = unpackInteger(raw_message);
+    auto xmax = unpackInteger(raw_message);
+    auto step = unpackInteger(raw_message);
+    finalize(raw_message);
+    dynamic_cast<IntTensor*>(r)->range(xmin, xmax, step);
+  } else if (isFloat(type)) {
+    auto xmin = unpackFloat(raw_message);
+    auto xmax = unpackFloat(raw_message);
+    auto step = unpackFloat(raw_message);
+    finalize(raw_message);
+    dynamic_cast<FloatTensor*>(r)->range(xmin, xmax, step);
+  } else {
+    throw std::runtime_error("expected a scalar type");
+  }
+}
+
+static void tensorRandperm(rpc::RPCMessage& raw_message) {
+  throw std::runtime_error("randperm is not available yet");
+}
+
+static void tensorSort(rpc::RPCMessage& raw_message) {
+  Tensor *rt = unpackRetrieveTensor(raw_message);
+  Tensor *ri = unpackRetrieveTensor(raw_message);
+  Tensor *tensor = unpackRetrieveTensor(raw_message);
+  int dimension = unpackInteger(raw_message);
+  int desc = unpackInteger(raw_message);
+  finalize(raw_message);
+  rt->sort(*ri, *tensor, dimension, desc);
+}
+
+static void tensorTopk(rpc::RPCMessage& raw_message) {
+  Tensor *rt = unpackRetrieveTensor(raw_message);
+  Tensor *ri = unpackRetrieveTensor(raw_message);
+  Tensor *tensor = unpackRetrieveTensor(raw_message);
+  long k = unpackInteger(raw_message);
+  int dimension = unpackInteger(raw_message);
+  int dir = unpackInteger(raw_message);
+  int sorted = unpackInteger(raw_message);
+  finalize(raw_message);
+  rt->topk(*ri, *tensor, k, dimension, dir, sorted);
+}
+
+static void tensorTril(rpc::RPCMessage& raw_message) {
+  Tensor *r = unpackRetrieveTensor(raw_message);
+  Tensor *t = unpackRetrieveTensor(raw_message);
+  long k = unpackInteger(raw_message);
+  finalize(raw_message);
+  r->tril(*t, k);
+}
+
+static void tensorTriu(rpc::RPCMessage& raw_message) {
+  Tensor *r = unpackRetrieveTensor(raw_message);
+  Tensor *t = unpackRetrieveTensor(raw_message);
+  long k = unpackInteger(raw_message);
+  finalize(raw_message);
+  r->triu(*t, k);
+}
+
+static void tensorCatArray(rpc::RPCMessage& raw_message) {
```

Comment:
Shouldn't this function do something? For now it just unpacks values and finalizes message.